src/CacheManager.Web/SystemWebCacheHandle`1.cs

```diff
@@ -86,6 +89,20 @@ public override bool Exists(string key, string region)
             return GetCacheItemInternal(key, region) != null;
         }
 
+        /// <inheritdoc />
+        protected override IEnumerable<string> AllKeys()
+        {
+            var keys = new List<string>();
+
+            var item = Context.Cache.GetEnumerator();
+            while (item.MoveNext())
+            {
+                keys.Add(item.Key.ToString());
```

Comment:
yield is not because of System.Web.Cache. As mentioned above, you should implement IEnumerable properly so that the iteration is handled per item and can stop at any point without ever iterating all keys. That must be implemented across all the methods to take effect